### PR TITLE
mosh: update to 20220318

### PR DESCRIPTION
**Pull request by jbenden**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (`x86_64`-`glibc`)

#### Notes
The upstream maintainer has seemingly vanished; therefore, a new release seems unlikely. As a workaround for the time being, switch to the git development branch.

### CODE REVIEW

File: srcpkgs/mosh/template [hunk 1]
```diff
@@ -1,19 +1,24 @@
 # Template file for 'mosh'
 pkgname=mosh
-version=1.3.2
-revision=24
+version=20220318
```
Comment (classabbyamp) [inline srcpkgs/mosh/template:3]: maybe a version like `1.3.2+20220318` or adding the commit hash or something would be better? If mosh gets another release in the future, I think that will break version comparison, as `20220318` is larger than `1`.

**Comment by leahneukirchen** [comment]
This is not true, the original maintainer is easily available but doesn't have time to make a new release (nor is there an urgent reason to). Someone else stepped up and is preparing for a 1.4 afaiu: https://mailman.mit.edu/pipermail/mosh-devel/2022-February/001598.html

In any way, the version should be kept such that an upstream 1.4 release can supersede this, but I don't see a compelling reason to switch to a Git snapshot currently.

File: srcpkgs/mosh/template (hunk 1 above)
Comment (jbenden) [reply srcpkgs/mosh/template]: Sounds good!

**Comment by jbenden** [comment]
I understand the defensive stance; however, but it's been years since the last release and a few good patches lay in the source. :)

**Pull request closed without merging**